src/Proto.Mailbox/Mailbox.cs

```diff
@@ -115,10 +115,9 @@ private Task RunAsync()
                     _stats[i].MailboxEmpty();
                 }
             }
-            return Task.FromResult(0);
         }
 
-        private bool ProcessMessages()
+        private async Task<bool> ProcessMessages()
```

Comment:
cc @cpx86 We should take a closer look at this as this part was highly optimized to avoid the async statemachine.
This will come at a pretty big perf penalty